File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int n = s.size();
        int mx = 0;

        unordered_map<char, int> mp;
        for(int i = 0; i < n; i++) {
            unordered_map<char, int> freq;
            for(int j = i; j < n; j++) {
                freq[s[j]]++;
                if(freq[s[j]] > 2)break;
                mx = max(mx, j-i+1);
            }
        }
        return mx;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        if len(s) <= 2:
            return len(s)
        else:
            new = 0
            for i in range(len(s)):
                subs = s[i:]
                for idx in range(len(subs)):
                    if subs[:idx].count(subs[idx]) == 2:
                        break
                    else:
                        longest = len(subs[:idx+1])
                        lon_str = subs[:idx+1]
                        print(idx, subs[:idx+1])
                if longest > new:
                    new = longest
            return new